core/EE_System.core.php

```diff
@@ -1207,6 +1221,11 @@ public function initialize()
     public function initialize_last()
     {
         do_action('AHEE__EE_System__initialize_last');
+        /** @var EventEspresso\core\domain\services\custom_post_types\RewriteRules $rewrite_rules */
+        $rewrite_rules = $this->loader->getShared(
+            'EventEspresso\core\domain\services\custom_post_types\RewriteRules'
+        );
+        $rewrite_rules->flushRewriteRules();
```

Comment:
why are you flushing the rewrite rules here?  This means rules are flushed on _every_ request and that will cause significant problems (not to mention being a very expensive thing to do on every request.

Reply:
No it only looks like that because of the name of the method. 
The actual code checks a WP option and if it is set to `true`, THEN it will flush the rewrite rules.
This kinda follows the "Tell Don't Ask" concept where the client code simply issues a command but the called service or entity actually decides whether to follow through with it or not.

Reply:
Understood.  But how come this is _added_ here?  There wasn't anything similar existing in this location before - is this replacing similar functionality that was elsewhere?

Reply:
this replaces the following code that used to be in the `EE_Register_CPTs` constructor

```
add_action('AHEE__EE_System__initialize_last', array(__CLASS__, 'maybe_flush_rewrite_rules'), 10);
```

Reply:
AHH, k thanks.  I should've realized that.